src/middlewares/FaclityValidation.js

```diff
@@ -13,9 +15,11 @@ const _validateFacility=async(req,res,next)=>{
     roomDetails:Joi.string().optional()
 
   });
-  const {error}=Joi.validate(req.body,_schema);
-  if(error) return res.status(422).json({message:error.details[0].message})
-  
+ 
+  if(req.method=='POST' || req.method=='PUT' ||req.method=='PATCH'){
+    const {error}=Joi.validate(req.body,_schema);
+    if(error) return res.status(422).json({message:error.details[0].message})
+  }
   const loggedinUser=req.headers.authorization;
   if(!loggedinUser) return res.status(403).json({message:"Provide you credentials"})
 
```

Comment:
1. Too many return statements, please consider refactoring, say, declare a result variable to which you conditionally assign the various responses and only return that variable after checking all conditions.
2. Why do you call the next() function twice?